### PR TITLE
[otbn/sca] Improve OTBN SCA capture setup

### PR DESCRIPTION
**Pull request by vrozic**
This PR contains two updates to improve OTBN ECC256 keygen capture setup:

- Mute ibex during the OTBN operation to reduce noise
- Move seed masking outside of the batch capture loop to prevent "fake leakage".

### CODE REVIEW

File: sw/device/sca/ecc256_keygen_serial.c [hunk 1]
```diff
@@ -60,6 +60,13 @@ enum {
    * Max number of traces per batch.
    */
   kNumBatchOpsMax = 256,
+  /**
+   * Number of cycles that Ibex should sleep to minimize noise during OTBN
+   * operations. Caution: This number should be chosen to provide enough time
+   * to complete the operation. Otherwise, Ibex might wake up while OTBN is
+   * still busy and disturb the capture.
+   */
+  kIbexOtbnSleepCycles = 800,
```
Comment (wettermo) [inline sw/device/sca/ecc256_keygen_serial.c:69]: Is this an estimated cycle count, or did you measure how many cycles the OTBN takes?

File: sw/device/sca/ecc256_keygen_serial.c [hunk 2]
```diff
@@ -219,24 +226,28 @@ static void ecc256_ecdsa_secret_keygen_batch(const uint8_t *data,
...
-      prng_rand_bytes((unsigned char *)batch_masks[i], kEcc256SeedNumBytes);
+      prng_rand_bytes((unsigned char *)batch_share1[i], kEcc256SeedNumBytes);
     } else {
       for (uint32_t j = 0; j < kEcc256SeedNumWords; ++j) {
-        batch_masks[i][j] = 0;
+        batch_share1[i][j] = 0;
       }
     }
+    for (uint32_t j = 0; j < kEcc256SeedNumWords; ++j) {
+      batch_share0[i][j] = batch_share1[i][j] ^ batch_share0[i][j];
```
Comment (jadephilipoom) [inline sw/device/sca/ecc256_keygen_serial.c:241]: nit:
```suggestion
      batch_share0[i][j] ^= batch_share1[i][j]
```

**Review by jadephilipoom — APPROVED** [review]
Seems like this partially addresses https://github.com/lowRISC/opentitan/issues/17338 (partial since there are other files to add sleep functionality to)

File: sw/device/sca/ecc256_keygen_serial.c (hunk 1 above)
Comment (vrozic) [reply sw/device/sca/ecc256_keygen_serial.c:69]: For now I've just selected `800` as a placeholder value. It is enough to mute ibex for the 3500 sample measurements that we are currently running.
Eventually, we plan to make this value programmable by the `capture.py` script, but I wouldn't want to do that in this PR.

**Review by abdullahvarici — APPROVED** [review]
Thanks @vrozic! I also tried it out and it is working. LGTM.

File: sw/device/sca/ecc256_keygen_serial.c (hunk 2 above)
Comment (vrozic) [reply sw/device/sca/ecc256_keygen_serial.c]: Thanks. This is fixed now.

File: sw/device/sca/ecc256_keygen_serial.c [hunk 3]
```diff
@@ -297,8 +314,14 @@ static void p256_ecdsa_gen_secret_key(const uint32_t *seed,
 static void p256_ecdsa_gen_keypair(const uint32_t *seed, const uint32_t *mask,
                                    uint32_t *d0, uint32_t *d1, uint32_t *x,
                                    uint32_t *y) {
+  // Compute first share of seed (seed ^ mask).
+  uint32_t seed0[kEcc256SeedNumWords];
```
[the remark this answers is not in the document]
Comment (vrozic) [reply sw/device/sca/ecc256_keygen_serial.c]: Agreed. It's fixed now

**Comment by vrozic** [comment]
Thanks @wettermo , @jadephilipoom and @abdullahvarici for the reviews. I've addressed the comments and added one small fix: Since seed masking was factored out of `p256_run_keygen()` it needed to be added back in `p256_ecdsa_gen_secret_key()` and `p256_ecdsa_gen_keypair()` before calling `p256_run_keygen()`.

**Pull request merged**